src/cairoUtilFuncGen/enumBoundCheck.ts

```diff
@@ -0,0 +1,72 @@
+import { CairoUtilFuncGenBase } from './base';
+import { CairoFunction } from './base';
+import {
+  EnumDefinition,
+  Expression,
+  VariableDeclaration,
+  UserDefinedTypeName,
+  FunctionCall,
+  IntType,
+} from 'solc-typed-ast';
+import assert = require('assert');
+import { createCairoFunctionStub, createCallToFunction } from '../utils/functionStubbing';
+import { typeNameFromTypeNode } from '../utils/utils';
+
+const INDENT = ' '.repeat(4);
+
+export class EnumBoundCheckGen extends CairoUtilFuncGenBase {
+  //Enum -> Assert Code
+  private generatedEnumBoundChecks: Map<string, CairoFunction> = new Map();
+
+  getGeneratedCode(): string {
+    return [...this.generatedEnumBoundChecks.values()].map((func) => func.code).join('\n\n');
+  }
+
+  gen(enumVarDec: VariableDeclaration, functionInput: Expression): FunctionCall {
+    const enumType = enumVarDec.vType;
+    assert(enumType instanceof UserDefinedTypeName);
+    const enumDef = enumType.vReferencedDeclaration;
+
+    assert(enumDef instanceof EnumDefinition);
+    const name = this.getOrCreate(enumDef);
+    const intType = new IntType(8, false);
```

Comment:
Is this the correct type for an `enum`, since it is an `int` behind the scenes?

Reply:
Usually, but not always, it can be a larger type if the enum has many members. I'd recommend using `enumToTypeInt` and `typeNameFromTypeNode` like in enumConverter to generate the appropriate typeName

Reply:
Yeah I agree. I also saw that the upper bound for enum members was much higher than int8.